### PR TITLE
Slightly clean up embassy HIL tests

### PR DESCRIPTION
**Pull request by bugadani**
Nothing major, just a bit of syntactic improvement.

### CODE REVIEW

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by bugadani** [comment]
Apparently my rustfmt doesn't agree with the CI's rustfmt 🤔

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/i2c.rs [hunk 1]
```diff
@@ -370,7 +370,7 @@ where
         sda: impl Peripheral<P = SDA> + 'd,
         scl: impl Peripheral<P = SCL> + 'd,
         frequency: HertzU32,
-        clocks: &Clocks<'d>,
```
Comment (Dominaezzz) [inline esp-hal/src/i2c.rs:373]: Is this correct? Won't the lifetime be inferred to something shorter that it should be?

Comment (bugadani) [reply esp-hal/src/i2c.rs:373]: I think it is, but regardless this change shouldn't be part of this PR.

**Pull request merged**